smokey/src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java

```diff
@@ -616,6 +615,25 @@ private ChassisSpeeds clampChassisSpeeds(ChassisSpeeds chassisSpeeds){
     MAX_ANGULAR_VELOCITY_RADIANS_PER_SECOND);
   }
 
+  private ChassisSpeeds discretize(ChassisSpeeds speeds) {
+    // a fudge factor to increase the size of the discretization correction. 
+    // other teams use [1..4]
+    double timeScaleFactor = 1.9; 
+    var desiredDeltaPose = new Pose2d(
+      speeds.vxMetersPerSecond * deltaTimeSeconds, 
+      speeds.vyMetersPerSecond * deltaTimeSeconds, 
+      new Rotation2d(speeds.omegaRadiansPerSecond * deltaTimeSeconds * timeScaleFactor)
+    );
+    var twist = new Pose2d().log(desiredDeltaPose);
```

Comment:
so this is the magic?

![image](https://github.com/Team4682CyBears/ChargedUp2023/assets/56608471/ead78619-b115-44d0-b60a-38b8e10b5452)

Reply:
Yep!